front/src/reducers/osrdconf/operationalStudiesConf/powerRestrictionReducer.ts

```diff
@@ -163,42 +163,31 @@ const powerRestrictionReducer = {
     action: PayloadAction<{
       firstRestriction?: PowerRestriction;
       secondRestriction: PowerRestriction;
+      endPosition: number;
       newFromPathStep: PathStep;
     }>
   ) {
-    const { firstRestriction, secondRestriction, newFromPathStep } = action.payload;
+    const { firstRestriction, secondRestriction, endPosition, newFromPathStep } = action.payload;
 
     // pathSteps should not be undefined or have null values
-    if (state.pathSteps && !state.pathSteps.some((pathStep) => !pathStep)) {
+    if (!state.pathSteps.some((pathStep) => !pathStep)) {
       let newPathSteps = [...state.pathSteps].map((step) => step!);
```

Comment:
Same for `resizeSegmentEndInput`
```suggestion
    if (state.pathSteps.every((pathStep) => !!pathStep)) {
      let newPathSteps = [...state.pathSteps];
```

Reply:
Nice catch, thanks :)

Reply:
Missed it, but seems like we can do the same l.249 for `cleanPowerRestrictionsCoveredByANewRange` ?